businessRules/donations.js

```diff
@@ -1,54 +1,64 @@
 const { v4 } = require('uuid');
 const { createDonation } = require('../models/donations');
-const { getAllIdeas } = require('../models/ideas');
-const { getIdeas } = require('../utils/storage');
+const { validateIdea } = require('../validations/ideas');
+const { validateSum } = require('../validations/donations.js');
+const { setUsers, getUsers } = require('../utils/storage');
+const { getUser } = require('../models/users');
+const { getIdea } = require('../models/ideas');
 
-function createDonation(firstName, sum, userId, ideaId) {
-    let ideas = getIdeas();
-    const idea = ideas.find((idea) => ideaId === idea.id);
+function createAnonymousUser(firstName) {
+    let users = getUsers();
 
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
-    }
-
-    const generateId = v4();
-    const anonymous = 'Anonymous';
+    const userId = v4();
+    const defaultFirstName = 'Anonymous';
+    const defaultUsername = 'none';
+    const defaultPassword = 'none';
+    const defaultLastName = 'none';
+    const defaultSession = 'none';
+    const defaultRole = 'guest';
 
-    firstName = firstName || anonymous;
-    userId = userId || generateId;
+    if (!firstName) {
+        firstName = defaultFirstName;
+    }
 
-    createDonation(firstName, sum, userId, ideaId);
+    const anonymousUser = {
+        id: userId,
+        picture: './img/default_userpic.webp',
+        userName: defaultUsername,
+        password: defaultPassword,
+        firstName,
+        lastName: defaultLastName,
+        session: defaultSession,
+        role: defaultRole,
+    };
+    const updatedUsers = [...users, anonymousUser];
+    setUsers(updatedUsers);
+    return anonymousUser;
 }
 
-function ideasDonationSum(ideaId) {
-    const ideas = getAllIdeas();
-    const idea = ideas.find((idea) => idea.id === ideaId);
-
-    if (!idea) {
-        throw new Error('Error. No idea with such ID found.');
-    }
-    return idea.totalDonationSum;
+function createDonationByUnregisteredUser(ideaId, firstName, sum) {
+    validateIdea(ideaId);
+    validateSum(sum);
+    const anonymousUser = createAnonymousUser(firstName);
+    createDonation(ideaId, anonymousUser.id, sum);
 }
 
-function ideasSumDifference(ideaId) {
-    const ideas = getAllIdeas();
-    const idea = ideas.find((idea) => idea.id === ideaId);
+function createDonationByRegisteredUser(userId, ideaId, sum) {
```

Comment:
For consistency, change parameters order to: `ideaId, userId, sum` - to match the order of params in `createDonationByUnregisteredUser` - where `ideaId` is the first param

Reply:
Understood that, seems logical to me. ) Fixed here and in index.js